server/src/main/java/io/crate/planner/node/ddl/CreateRolePlan.java

```diff
@@ -53,19 +68,49 @@ public StatementType type() {
     public void executeOrFail(DependencyCarrier dependencies,
                               PlannerContext plannerContext,
                               RowConsumer consumer,
-                              Row params, SubQueryResults subQueryResults) throws Exception {
-        SecureHash newPassword = UserActions.generateSecureHash(
+                              Row params,
+                              SubQueryResults subQueryResults) throws Exception {
+
+        Map<String, Object> properties = parse(
             createRole.properties(),
-            params,
             plannerContext.transactionContext(),
-            plannerContext.nodeContext());
+            plannerContext.nodeContext(),
+            params,
+            subQueryResults
+        );
+        SecureHash newPassword = UserActions.generateSecureHash(properties);
 
         if (createRole.isUser() == false && newPassword != null) {
             throw new UnsupportedOperationException("Creating a ROLE with a password is not allowed, " +
                                                     "use CREATE USER instead");
         }
 
-        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword)
+        // Map is always not null to simplify streaming.
```

Comment:
also invalid comment?

Reply:
already removed in https://github.com/crate/crate/pull/15640/commits/5bd489a2379ac72a9effbea9d881e1f5dc685268#diff-05ff3ea28f321d32053d4ba4043a7006962031aed40b321d33df54ba72620317L88